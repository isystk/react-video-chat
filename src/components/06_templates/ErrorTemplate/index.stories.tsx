import { Meta, Story } from '@storybook/react'
import React from 'react'
import ErrorTemplate from './index'
import MainService from '@/services/main'
import { Context } from '@/components/05_layouts/HtmlSkeleton'

export default {
  title: '06_templates/ErrorTemplate',
  component: ErrorTemplate,
} as Meta

const Template: Story = (props) => {
  const main = new MainService(() => ({}))
  return (
    <Context.Provider value={main}>
      <ErrorTemplate statusCode="404" />
    </Context.Provider>
  )
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
