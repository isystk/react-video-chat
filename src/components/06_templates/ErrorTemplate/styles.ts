import { css, keyframes } from '@emotion/css'

export const container = css`
  text-align: center;
`
