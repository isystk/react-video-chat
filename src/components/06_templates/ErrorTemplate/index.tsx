import HtmlSkeleton, {
  Context,
  HtmlSkeletonProps,
  Title,
} from '@/components/05_layouts/HtmlSkeleton'
import { connect } from '@/components/hoc'
import React, { useContext } from 'react'
import { ContainerProps } from 'types'
import * as styles from './styles'
import { Container, Typography } from '@mui/material'

/** ErrorTemplate Props */
export type ErrorTemplateProps = Omit<HtmlSkeletonProps, 'children'> & {
  main
  statusCode: string
}
/** Presenter Props */
export type PresenterProps = ErrorTemplateProps

/** Presenter Component */
const ErrorTemplatePresenter: React.FC<PresenterProps> = ({
  statusCode,
  main,
  ...props
}) => (
  <HtmlSkeleton>
    <Title>Error</Title>
    <Container className={styles.container}>
      <div>
        <Typography component="h1" variant="h5">
          {statusCode} エラーが発生しました。
        </Typography>
      </div>
    </Container>
  </HtmlSkeleton>
)

/** Container Component */
const ErrorTemplateContainer: React.FC<
  ContainerProps<ErrorTemplateProps, PresenterProps>
> = ({ presenter, main, ...props }) => {
  return presenter({ main, ...props })
}

/** ErrorTemplate */
export default connect<ErrorTemplateProps, PresenterProps>(
  'ErrorTemplate',
  ErrorTemplatePresenter,
  ErrorTemplateContainer
)
