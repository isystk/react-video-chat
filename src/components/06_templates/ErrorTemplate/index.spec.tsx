import React from 'react'
import renderer from 'react-test-renderer'
import ErrorTemplate from './index'
import '@testing-library/jest-dom/extend-expect'
import { Provider } from 'react-redux'
import { applyMiddleware, createStore } from 'redux'
import thunk from 'redux-thunk'
import reducers from '@/stores'

describe('ErrorTemplate', () => {
  it('Match Snapshot', () => {
    const enhancer = applyMiddleware(thunk)
    const store = createStore(reducers, enhancer)

    const component = renderer.create(
      <Provider store={store}>
        <ErrorTemplate statusCode={404} />
      </Provider>
    )
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
