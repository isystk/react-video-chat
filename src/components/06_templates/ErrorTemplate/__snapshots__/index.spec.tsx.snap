// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorTemplate Match Snapshot 1`] = `null`;
